src/http/errors.ts

```diff
@@ -1,23 +1,43 @@
-import makeError from 'make-error'
+import {BaseError} from 'make-error'
 
-function ClientError(res) {
-  const props = extractErrorProps(res)
-  ClientError.super.call(this, props.message)
-  Object.assign(this, props)
+import {ErrorProps} from '../types'
+
+/** @public */
+export class ClientError extends BaseError {
+  response: ErrorProps['response']
+  statusCode: ErrorProps['statusCode'] = 400
+  responseBody: ErrorProps['responseBody']
+  details: ErrorProps['details']
+
+  constructor(res: any) {
```

Comment:
Good idea, an alias would let us remove `--quiet` from the linter command and still avoid all the noise

Reply:
There are no typings here because it wasn't defined in the original https://github.com/sanity-io/client/blob/main/sanityClient.d.ts#L884 file.
Fixing all of them is out of scope for this PR, I suggest doing that in follow ups

Reply:
Makes perfect sense!

Reply:
@bjoerge I've replaced all the explicit `any` instances with the `FIXME` type, so we can follow up and incrementally fix them and improve the typings after v5 is released 🙇